configs/projects/_shared-agent-components.rb

```diff
@@ -48,10 +48,9 @@
 proj.component 'rubygem-gettext'
 proj.component 'rubygem-fast_gettext'
 
-if platform.is_windows? || platform.is_solaris?
-  proj.component 'rubygem-ffi'
-  proj.component 'rubygem-minitar'
-end
+proj.component 'rubygem-ffi' if platform.is_windows? || platform.is_solaris? ||
+                                (platform.is_aix? && proj.settings[:is_agent_master])
```

Comment:
Instead of making this a shared component you can add it only in `agent-runtime-master` project. In this way we won't need the `is_agent_master` setting